neptunecontrib/monitoring/optuna.py

```diff
@@ -13,36 +13,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import warnings
 
 import neptune
-from neptunecontrib.monitoring.utils import pickle_and_send_artifact
 
+from neptunecontrib.api import log_chart, pickle_and_log_artifact
 
-def NeptuneMonitor(experiment=None):
+
+def NeptuneCallback(experiment=None, log_charts=False, log_study=False, params=None):  # pylint: disable=W0621
```

Comment:
Why a function and not a class with `__call__`?

Reply:
I actually spent a good 15 minutes on checking how this even works (since Optuna doesn't know anything about a `monitor`) before noticing `return monitor` :smile:

Reply:
Looking at optuna callbacks using __call__ is the way they do it so I am on it.